packages/http-core/src/request/request.js

```diff
@@ -46,6 +46,23 @@ function doRequest(options) {
 
       if (response) {
         response.options = options;
+
+        // I'm not sure why this line is necessary. request seems to be creating
+        // buffers that aren't Buffers.
+        if (options.responseType === `buffer` && response.body.type === `Buffer` && !Buffer.isBuffer(response.body)) {
+          response.body = new Buffer(response.body);
+        }
```

Comment:
If it looks like a buffer and says it's a buffer, but it's not a buffer, then make it a buffer :)